src/pyosmeta/models/github.py

```diff
@@ -0,0 +1,129 @@
+"""
+Models for Github API requests autogenerated from JSON schema,
+and then deduplicated and trimmed for just specifying what we use here.
+
+eg. given the Issue schema at https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#get-an-issue
+Using `datamodel-code-generator` ( https://docs.pydantic.dev/latest/integrations/datamodel_code_generator/ )
+
+```
+datamodel-codegen --input issue_schema.json --input-file-type jsonschema --output issue_models.py
+```
+"""
+
+# generated by datamodel-codegen:
+#   filename:  issue_schema.json
+#   timestamp: 2024-07-02T23:19:14+00:00
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, List, Literal, Optional, Union
+
+from pydantic import AnyUrl, BaseModel, ConfigDict, Field
+
+
+class User(BaseModel):
```

Comment:
@sneakers-the-rat we do this in stravalib but have a bot to grab things. Can you answer a few questions for me about this (it's awesome and i want to learn more & document it correctly). 

1 should we add datamodel-codegen to optional dependencies? 
2. where is issue_schema .json coming from? with strava there is a swagger.json file that we grab online. 
3. did you manually trim things to map to our workflow here?

thank you again.

Reply:
@lwasser sorry I missed this question -
1) nah, it was just a one-off to create the model from the JSON schema, and then I heavily modified and trimmed it anyway.
2) the URL I got it from is in the docstring :)
3) yep! Basically I took out all the fields that we weren't/probably wouldnt use and set `extra = "allow"` in the model config so that those extra fields would be there/wouldnt throw a validation error, but youd get a static analysis warning if you tried to access them (hinting they should be added to the model formally). I also simplified the various `User` models - each of the blank models that inherit from the `User` class now were just duplicates. And I also made some fields optional I blv to let the tests run. But basically those models in the `github` module shouldnt be changed (unless github's API changes) - we just want those to reflect what comes back from the API, and the models we use internally in this package are separate.